middleware/cmd/middleware/main.go

```diff
@@ -74,6 +83,25 @@ func main() {
 	}
 	log.Println("--------------- Started middleware --------------")
 
+	// Check for available hsm firmware update on middleware boot and update if available
+	if hsmFirmware != nil {
+		updateAvailable, err := middleware.HSMUpdateAvailable(hsm)
+		if err != nil {
+			log.Printf("Failed to fetch HSM version/update information: %s. Proceeding normally", err)
+		}
+		if updateAvailable {
+			err = hsm.UpdateFirmware(*hsmFirmwareFile)
+			if err != nil {
+				log.Printf("Failed to update HSM firmware: %s", err)
+			} else {
+				err = middleware.BootHSMFirmware(hsm)
```

Comment:
This is a bit hard to follow. I'd just call `hsmFirmware, err := hsm.WaitForFirmware()` instead, and move everything up to line 52 to group all the HSM init related things.

Reply:
Moved to the middleware instead of main as discussed offline. Will finish the WIP tomorrow